tests/components/moehlenhoff_alpha2/test_config_flow.py

```diff
@@ -0,0 +1,91 @@
+"""Test the moehlenhoff_alpha2 config flow."""
+import asyncio
+from unittest.mock import patch
+
+from homeassistant.components.moehlenhoff_alpha2.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+
+async def mock_fetch_static_data(self):
+    """Mock moehlenhoff_alpha2.Alpha2Base._fetch_static_data."""
+    self.static_data = {
+        "Devices": {"Device": {"ID": "fake_id", "NAME": "fake_base_name"}}
+    }
+
+
+async def test_duplicate_error(hass: HomeAssistant):
+    """Test that errors are shown when duplicates are added."""
+    with patch(
+        "moehlenhoff_alpha2.Alpha2Base._fetch_static_data", mock_fetch_static_data
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data={"host": "fake_host"}
+        )
+        assert result["title"] == "fake_base_name"
+        assert result["type"] == "create_entry"
+        assert not result["result"].unique_id
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": "user"}, data={"host": "fake_host"}
+        )
+        assert result["type"] == "abort"
+        assert result["reason"] == "already_configured"
+
+
+async def test_user(hass: HomeAssistant):
+    """Test starting a flow by user."""
+
+    with patch(
+        "moehlenhoff_alpha2.Alpha2Base._fetch_static_data", mock_fetch_static_data
+    ):
+        result = await hass.config_entries.flow.async_init(
```

Comment:
We want to patch the integration setup function on all create entry results to avoid setting up the whole integration. See the scaffold test template for example.